src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -179,37 +182,127 @@ class ExternalDatabaseManagementService(
+
+            //delete tables from postgres
+            val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+            assemblerConnectionManager.connect(dbName).let { dataSource ->
+                dataSource.connection.createStatement().use { stmt ->
+                    stmt.execute("DROP TABLE $tableName")
+                }
+            }
+        }
+
+        //delete securable object
+        val tableIds = tableIdByFqn.values.toSet()
+        deleteOrganizationExternalDatabaseTableObjects(tableIds)
+    }
+
+    fun deleteOrganizationExternalDatabaseTable(orgId: UUID, tableFqnToId: Pair<String, UUID>) {
+        deleteOrganizationExternalDatabaseTables(orgId, mapOf(tableFqnToId))
+    }
+
+    fun deleteOrganizationExternalDatabaseTableObjects(tableIds: Set<UUID>) {
         tableIds.forEach {
-            val aclKey = AclKey(orgId, it)
+            val aclKey = AclKey(it)
             authorizationManager.deletePermissions(aclKey)
             securableObjectTypes.remove(aclKey)
             aclKeyReservations.release(it)
-            val columnIdsByTableId = mapOf(it to organizationExternalDatabaseColumns.keySet(belongsToTable(it)))
-            deleteOrganizationExternalDatabaseColumns(orgId, columnIdsByTableId)
         }
+        organizationExternalDatabaseTables.executeOnEntries(DeleteOrganizationExternalDatabaseTableEntryProcessor(), idsPredicate(tableIds))
     }
 
-    fun deleteOrganizationExternalDatabaseTable(orgId: UUID, tableId: UUID) {
-        deleteOrganizationExternalDatabaseTables(orgId, setOf(tableId))
-    }
+    fun deleteOrganizationExternalDatabaseColumns(orgId: UUID, columnsByTable: Map<Pair<String, UUID>, Map<String, UUID>>) {
+        columnsByTable.forEach { (tableFqnToId, columnIdsByFqn) ->
+            if (columnIdsByFqn.isEmpty()) return@forEach
 
-    fun deleteOrganizationExternalDatabaseColumns(orgId: UUID, columnIdsByTableId: Map<UUID, Set<UUID>>) {
-        columnIdsByTableId.forEach { (tableId, columnIds) ->
-            organizationExternalDatabaseColumns.executeOnEntries(DeleteOrganizationExternalDatabaseColumnsEntryProcessor(), idsPredicate(columnIds))
+            val tableName = getNameFromFqnString(tableFqnToId.first)
+            val tableId = tableFqnToId.second
+            val columnNames = columnIdsByFqn.keys.map { getNameFromFqnString(it) }.toSet()
+            val columnIds = columnIdsByFqn.values.toSet()
+
+            //delete columns from postgres
+            val dropColumnsSql = createDropColumnSql(columnNames)
+            val dbName = PostgresDatabases.buildOrganizationDatabaseName(orgId)
+            assemblerConnectionManager.connect(dbName).let {
+                it.connection.createStatement().use { stmt ->
+                    stmt.execute("ALTER TABLE $tableName $dropColumnsSql")
+                }
+            }
+
+
+
+            //delete securable objects
             columnIds.forEach {
```

Comment:
Shouldn't this code block be just a call to `deleteOrganizationExternalDatabaseColumnObjects` ?

Reply:
yea it should, forgot to do that bit. ty for catching!